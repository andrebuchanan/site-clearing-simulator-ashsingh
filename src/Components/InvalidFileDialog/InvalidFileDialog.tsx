import React from 'react'
import Button from '@mui/material/Button'
import Dialog from '@mui/material/Dialog'
import DialogActions from '@mui/material/DialogActions'
import DialogContent from '@mui/material/DialogContent'
import DialogContentText from '@mui/material/DialogContentText'

type InvalidFileDialogProps = {
  isOpen: boolean
  handleClick: () => void
}

export const InvalidFileDialog = ({
  isOpen,
  handleClick
}: InvalidFileDialogProps) => {
  return (
    <Dialog
      open={isOpen}
      onClose={handleClick}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          The file you have uploaded is invalid. Please reupload file.
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClick} autoFocus data-testid="close-dialog">
          Ok
        </Button>
      </DialogActions>
    </Dialog>
  )
}

export default InvalidFileDialog
