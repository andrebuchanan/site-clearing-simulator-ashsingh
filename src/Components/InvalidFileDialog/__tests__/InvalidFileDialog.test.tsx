import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import InvalidFileDialog from '../InvalidFileDialog'

describe('Quit Dialog', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <InvalidFileDialog isOpen handleClick={jest.fn()} />
    )
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText(/Please reupload file/i)).toBeVisible()
  })

  it('closes dialog', async () => {
    const mockFn = jest.fn()
    render(<InvalidFileDialog isOpen handleClick={mockFn} />)
    await waitFor(() => userEvent.click(screen.getByTestId('close-dialog')))
    expect(mockFn).toHaveBeenCalled()
  })
})
