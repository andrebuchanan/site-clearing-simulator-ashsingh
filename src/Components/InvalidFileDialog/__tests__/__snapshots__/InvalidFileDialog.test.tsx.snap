// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quit Dialog renders correctly 1`] = `<DocumentFragment />`;
